### ISSUE TITLE
add brake handling to pomodoro tracker

### ISSUE THREAD

**Issue by grzegorz-jazurek** [toggl/toggl-button#597]
It would be great if pomodoro tracker would have break handling

- after work phase we should count break phase (default time 5min)
- when break phase finish trigger alarm
- break phase can be editable in settings

**Comment by Mock1** [comment]
I think this is important too. I am trying to get in a groove with Toggl, and specifically want to drive work with a pomodoro method. I was disappointed to find the "break" was just dead time, and no timer or reminder to get back to work. Idle tracking could do it, but it would be better if it explicitly did real breaks.
I have no idea how long it's been since I finished my last pomodoro cycle...
The point of pomodoro is to get into a cadence, right?

**Comment by IndrekV** [comment]
More user feedback:

>Please, add Pomodoro 5 min break count down with a green Toggl button (to show it is a break)!!! and prompt to "Continue the latest task" or "Start a new one" in the end of the 5 min break. Furthermore, show notification that 25 minutes has finished and 5 min break has finished.
Thank you!

**Comment by Stvad** [comment]
Yep pause handling is very important part of Pomodoro workflow.
As an interesting exercise, you can create a Zap on http://zapier.com (ifttt analog) that would detect new entry creation in Toggle then wait some defined time and then trigger push notification on your phone (or something else).
The problem with this one is that it only runs once in 15 minutes (or 5 if you have premium) so it's not very useful.

**Comment by ARudiuk** [comment]
I recently started using Toggle and would very much like to have the functionality to track break time!

**Comment by karthik-vm** [comment]
Thanks for the great work. It will also be nice if a long break is added after 4 pomodoros. It is an integral part of pomodoro technique. The user should also be able to customize the break timings. Hope it is added soon.

**Comment by amitness** [comment]
Yes, this feature is essential. I'm using a pomodoro extension for now but once this is integrated, I'll come back to Toggl.

**Comment by plentylife** [comment]
Yes please, I also want this feature

**Comment by semka** [comment]
Really miss this feature

**Comment by mgehlen** [comment]
I agree, pomodoro technique does not really work when the break is not included correctly. I think the pomodoro feature cannot be used sufficiently as long as this is missing. But thanks to @Stvad for the zapier workaround, maybe I give that a try.

**Comment by MisterVulcan** [comment]
Ditto, and hello from 2018.

**Comment by amrtn** [comment]
Are you accepting PRs that address this issue?

**Comment by Stvad** [comment]
@amrtn They've accepted PR for original Pomodoro implementation ;)

**Comment by chrismilleresp** [comment]
I'm testing Toggl and I didn't find the Countdown option for the pomodoro timer. That would be nice!

**Comment by hidenori-shinohara** [comment]
This feature would be great!

**Comment by tcrammond** [comment]
@amrtn I'm sorry it's been a while since your comment, but yes we would definitely consider PRs that implement this feature. It would need to include (minimal) configuration options in the pomodoro timer settings.

**Comment by icklecoder** [comment]
Not sure how you vote on GH, but a big plus one here.  Love the chrome extension but have to kind of track my break time, so this would be amazing.
So automatic switch to break after the duration, long break after 4 would be just amazing, but the short break is the big win here.

**Comment by sterid** [comment]
2019 nothing is changed 👎

**Comment by francesco-vision** [comment]
Yes, break tracking is part of the Pomodoro technique, so Toggl button is not fully implementing the pomodoro technique yet, It would be necessary to do it according to me. We are in 2020 and nothing has changed, not very promising.

**Comment by AlainBit** [comment]
Would be good to get a sound notification when a break is completed as well as a pause functionality in case a pomodoro is briefly interrupted. Pomodone has both these features

**Comment by AlainBit** [comment]
It seems incredible that the coding does not allow for the Pomodoro timer to be paused: https://github.com/toggl/toggl-button/issues/297 Not sure why this was closed.

**Comment by bytrangle** [comment]
It may be bewildering for users to see that this issue has been ongoing for 5 years, but I can understand that. Integrating a Pomodoro tracker would introduce a lot of moving pieces that involves the backend when Toggle Track was never a Pomodoro timer to begin with.

That being said, I'd like to see interested users fleshing out your expectation of a Pomodoro feature in Toggle Tracker. 

1. What happens when a work phase completes? Does Toggle run a *Pomodoro break* automatically or does the user have to click a *Take a break* button to run it?
2. What happens when a break finishes? Does Toggle run the previous time entry again, or the user has to manually start it?

**Comment by rosyvs** [comment]
@bytrangle great question, I think that both options (auto roll-over vs manual start) make sense in the Pomodoro workflow for different people/situations. So it would be great to have this as an option set by the user - auto vs manual. 

My 2 cents to add: the existing Pomodoro timer functionality that Toggl has could benefit from simple changes to the messages that occur at the end of each work session/break. Note I use the Windows app, not sure if this also applies to the Chrome plugin etc. 

Currently, I see "Continue" or "Start New" as my two options when I get my pomodoro notification.

I can never remember which one is the right one to select ... does "Continue" extend the current break/ work session? Or does it roll over to the next work/break (if the current activity is break/work respectively) ?

Changing this language to "Extend current work session" and "Extend current break" vs "start work session" / "start break" would make it a whole lot clearer.

**Comment by bytrangle** [comment]
@rosyvs 
Oh, you're talking about the desktop app, which is close-sources. This Github is for the Toggl browser extension, which is open-source. They're almost like different apps.

As far as the Mac app is concerned, I only get a *Continue* option, which in fact starts a new work session.
I know this issue has been ongoing for 5 years, and I used to wonder why what seemed to be an easy feature hadn't been implemented yet.

Now that I've become a developer, I got it. Tracking time is an inherently difficult thing to program, as there are many user actions that developers have to account for. Everyone wants to a more refined handle of break times, yet when I probe further, no one replies. This Toggl extension is an open-source project, so I can understand why contributors hesitate to take upon it.

Regarding your wish for changing the wording on the Pomodoro notification, I'm afraid it is something that no contributor will take on. Contributions fall into two categories: bug fix or feature enhancement.  All contributions require approval from the project owner, but feature enhancements will always go under more scrutiny because it can break the application.

Changing wordings on an app notification seems trivial, but it affects the user experience and branding of an app and probably require approval from more stakeholders. Many contributors here are not employees of Toggl. If you want this feature so bad, you can contact their customer support directly. They are quite responsive from my experience.

**Comment by DivineMK** [comment]
I wonder if break handler is still considered.

**Comment by bytrangle** [comment]
Rereading the code base, the background script that contains the implementation of the Pomodoro feature has been removed (made close source). That rules out any chance of making PR for any outside contributors. Is there any value to keeping this issue open?
